### DISCUSSION TITLE
Installing Argo CD - what do i miss?

### DISCUSSION THREAD

**Discussion by ggruening** [kube-hetzner/terraform-hcloud-kube-hetzner#1452] [Q&A]
Hello everyone,

maybe it's a beginner's question - but I only relied on the resource in this repo. And it doesn't work straight away with those, so it might be of more interest if we discuss a few tips here and adjust the README at the end if necessary...

I'd like to install **Argo CD** using Kustomize and a helm chart. I tried the following:

- According to the "[Adding Extras](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner?tab=readme-ov-file#adding-extras)" chapter in the README, I created **a folder called "extra-manifests/helm-charts/argocd"**.

- I copied the three files from [**examples/kustomization_user_deploy/helm-chart**](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/tree/da24fd260b060038630a72b121f06288bc6b8e56/examples/kustomization_user_deploy/helm-chart) into it and only adjusted the domain in helm-chart.yaml.tpl.

- Among the examples in the README there is a note that you should extend the kube.tf for Argo CD. I did that:
```
  extra_kustomize_deployment_commands = <<-EOT
    kubectl -n argocd wait --for condition=established --timeout=120s crd/appprojects.argoproj.io
    kubectl -n argocd wait --for condition=established --timeout=120s crd/applications.argoproj.io
    kubectl apply -f /var/user_kustomize/argocd-projects.yaml
    kubectl apply -f /var/user_kustomize/argocd-application-argocd.yaml
    ...
  EOT
```

But running `terraform apply` leads to:

```
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): Applying user kustomization...
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): error: unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory '/var/user_kustomize'
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "appprojects.argoproj.io" not found
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "applications.argoproj.io" not found
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): error: the path "/var/user_kustomize/argocd-projects.yaml" does not exist
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): error: the path "/var/user_kustomize/argocd-application-argocd.yaml" does not exist
module.kube-hetzner.null_resource.kustomization_user_deploy[0] (remote-exec): /tmp/terraform_446928275.sh: line 9: ...: command not found
╷
│ Error: remote-exec provisioner error
│ 
│   with module.kube-hetzner.null_resource.kustomization_user_deploy[0],
│   on .terraform/modules/kube-hetzner/kustomization_user.tf line 48, in resource "null_resource" "kustomization_user_deploy":
│   48:   provisioner "remote-exec" {
│ 
│ error executing "/tmp/terraform_446928275.sh": Process exited with status 127
╵

```

What am I missing?

Thanks for a hint,
Gregor

**Comment by vaceslav** [comment]
File names should end with .tpl, such as "argocd-application-argocd.yaml.tpl". Does your file follow this naming convention?

**Reply by vaceslav** [reply]
I just tested it on my development system: all files should be in the “extra-manifests” directory, not in “helm-chart”.

**Comment by HendrikLevering** [comment]
I use argocd. I set it up directly in a terraform file with helm provider:

`resource "helm_release" "argocd" {
  name             = "argocd"
  repository       = "https://argoproj.github.io/argo-helm"
  chart            = "argo-cd"
  version          = "5.55.0" # Specify the version you want to install
  namespace        = "argocd"
  create_namespace = true
  set {
    name  = "redis-ha.enabled"
    value = "false" # Set the service type to LoadBalancer or NodePort as needed
  }

  set {
    name  = "controller.replicas"
    value = "1"
  }
  set {
    name  = "server.replicas"
    value = "1"
  }
  set {
    name  = "repoServer.replicas"
    value = "2" # Set to "true" to enable Ingress
  }
  set {
    name  = "applicationSet.replicas"
    value = "2" 
  }
}`
